File: UseCases/src/codes/seanhenry/mockgenerator/util/AppendStringDecorator.java
package codes.seanhenry.mockgenerator.util;

public class AppendStringDecorator extends StringDecorator {

  private String suffix;

  public AppendStringDecorator(StringDecorator decorator, String suffix) {
    super(decorator);
    this.suffix = suffix;
  }

  @Override
  protected String decorate(String string) {
    if (string == null || string.isEmpty()) return "";
    if (suffix == null || suffix.isEmpty()) return string;
    return string + suffix;
  }
}

File: UseCases/src/codes/seanhenry/mockgenerator/util/AppendStringDecorator.kt
package codes.seanhenry.mockgenerator.util

class AppendStringDecorator(decorator: StringDecorator?, private val suffix: String) : StringDecorator(decorator) {

  public override fun decorate(string: String): String {
    if (string == null || string.isEmpty()) return ""
    return if (suffix == null || suffix.isEmpty()) string else string + suffix
  }
}
